source/base/subscriptor.cc

```diff
@@ -62,39 +62,57 @@ Subscriptor::Subscriptor (const Subscriptor &)
 {}
 
 
+
+#ifdef DEAL_II_WITH_CXX11
+Subscriptor::Subscriptor (Subscriptor &&subscriptor)
+  :
+  counter(0),
+  object_info (subscriptor.object_info)
+{
+  subscriptor.check_no_subscribers();
+  subscriptor.counter = 0;
+  subscriptor.counter_map.clear();
+}
+#endif
+
+
+
 Subscriptor::~Subscriptor ()
 {
-  // check whether there are still
-  // subscriptions to this object. if
-  // so, output the actual name of
-  // the class to which this object
-  // belongs, i.e. the most derived
-  // class. note that the name may be
-  // mangled, so it need not be the
-  // clear-text class name. however,
-  // you can obtain the latter by
-  // running the c++filt program over
-  // the output.
+  check_no_subscribers();
+  counter = 0;
+  counter_map.clear();
+
```

Comment:
No need for these last two lines - the enclosing object will be destroyed in a second anyway,  which will take care of these variables anyway.